runelite-api/src/main/java/net/runelite/api/Diary.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019 MrGroggle
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.api;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * An enumeration of Diaries.
+ */
+@RequiredArgsConstructor
+@Getter
+public enum Diary
+{
+	ARDOUGNE_EASY("Ardougne Easy", Varbits.DIARY_ARDOUGNE_EASY),
+	ARDOUGNE_MEDIUM("Ardougne Medium", Varbits.DIARY_ARDOUGNE_MEDIUM),
+	ARDOUGNE_HARD("Ardougne Hard", Varbits.DIARY_ARDOUGNE_HARD),
+	ARDOUGNE_ELITE("Ardougne Elite", Varbits.DIARY_ARDOUGNE_ELITE),
+	DESERT_EASY("Desert Easy", Varbits.DIARY_DESERT_EASY),
+	DESERT_MEDIUM("Desert Medium", Varbits.DIARY_DESERT_MEDIUM),
+	DESERT_HARD("Desert Hard", Varbits.DIARY_DESERT_HARD),
+	DESERT_ELITE("Desert Elite", Varbits.DIARY_DESERT_ELITE),
+	FALADOR_EASY("Falador Easy", Varbits.DIARY_FALADOR_EASY),
+	FALADOR_MEDIUM("Falador Medium", Varbits.DIARY_FALADOR_MEDIUM),
+	FALADOR_HARD("Falador Hard", Varbits.DIARY_FALADOR_HARD),
+	FALADOR_ELITE("Falador Elite", Varbits.DIARY_FALADOR_ELITE),
+	FREMENNIK_EASY("Fremennik Easy", Varbits.DIARY_FREMENNIK_EASY),
+	FREMENNIK_MEDIUM("Fremennik Medium", Varbits.DIARY_FREMENNIK_MEDIUM),
+	FREMENNIK_HARD("Fremennik Hard", Varbits.DIARY_FREMENNIK_HARD),
+	FREMENNIK_ELITE("Fremennik Elite", Varbits.DIARY_FREMENNIK_ELITE),
+	KANDARIN_EASY("Kandarin Easy", Varbits.DIARY_KANDARIN_EASY),
+	KANDARIN_MEDIUM("Kandarin Medium", Varbits.DIARY_KANDARIN_MEDIUM),
+	KANDARIN_HARD("Kandarin Hard", Varbits.DIARY_KANDARIN_HARD),
+	KANDARIN_ELITE("Kandarin Elite", Varbits.DIARY_KANDARIN_ELITE),
+	KARAMJA_EASY("Karamja Easy", Varbits.DIARY_KARAMJA_EASY),
+	KARAMJA_MEDIUM("Karamja Medium", Varbits.DIARY_KARAMJA_MEDIUM),
```

Comment:
This diary varbit may have changed. My main which has completed this diary, has a varbit value of 2 for 3599 (`DIARY_KARAMJA_MEDIUM`), and my pure which has not completed it has a value of 1. Regardless, I'm seeing it crossed out on either account when logging in despite having the requirements met on my main account.

Reply:
It's also possible this is simply loading with the wrong value when the agility icons are being added. Regardless of the cause, this is the only case which seems to be giving incorrect results on both of my accounts.

Reply:
Good catch, thank you for looking into it this thoroughly. As with the quest start icons, I believe this could be related to the client not yet being logged in when the icons are first initialised, which then doesn't update after login correctly. I'll do some testing on my alt accounts and see if I can reproduce it, and aim to get a fix in place.

Reply:
The issue of switching accounts seems to be fixed, but I'm still getting the incorrect shilo village results from my review comment.

Reply:
It's looking like the varbit for Karamja Medium is incorrect. In the inspector the value for mine reads 1 on login, but I've not completed it yet.